vunit/sim_if/xcelium.py

```diff
@@ -0,0 +1,694 @@
+        Create the cds.lib file in the output directory if it does not exist
+        """
+        cds_root_virtuoso = self.find_cds_root_virtuoso()
+
+        if cds_root_virtuoso is None:
+            contents = """\
+## cds.lib: Defines the locations of compiled libraries.
+softinclude {0}/tools/inca/files/cds.lib
+# needed for referencing the library 'basic' for cells 'cds_alias', 'cds_thru' etc. in analog models:
+# NOTE: 'virtuoso' executable not found!
+# define basic ".../tools/dfII/etc/cdslib/basic"
+define work "{1}/libraries/work"
+
+# user defined cds by changing: global.xcelium.xrun_cdslib
+""".format(
+                self._cds_root_xrun, self._output_path
+            )
+        else:
+            contents = """\
+## cds.lib: Defines the locations of compiled libraries.
+softinclude {0}/tools/inca/files/cds.lib
+# needed for referencing the library 'basic' for cells 'cds_alias', 'cds_thru' etc. in analog models:
+define basic "{1}/tools/dfII/etc/cdslib/basic"
+define work "{2}/libraries/work"
+
+# user defined cds by changing: global.xcelium.xrun_cdslib
+""".format(
+                self._cds_root_xrun, cds_root_virtuoso, self._output_path
+            )
+
+
+        # add our cdslib customization
+        xrun_cdslibs = self.get_global_compile_option("xcelium.xrun_cdslib")
+        for xrun_cdslib in xrun_cdslibs:
+            contents += xrun_cdslib + "\n"
+
+
+        contents += "\n"
+        contents += "# project libraries \n"
+        write_file(self._cdslib, contents)
+
+    def set_global_compile_options(self, global_options: Dict):
+        self._global_compile_options = global_options
+
+    def get_global_compile_option(self, name):
+        """
+        Return a copy of the compile option list
+        """
+        if name not in self._global_compile_options:
+            self._global_compile_options[name] = []
+
+        return copy(self._global_compile_options[name])
+
+    def setup_library_mapping(self, project: Project):
+        """
+        Compile project using vhdl_standard
+        """
+        self._create_cdslib()
+        mapped_libraries = self._get_mapped_libraries()
+
+        for library in project.get_libraries():
+            self._libraries.append(library)
+            self.create_library(library.name, library.directory, mapped_libraries)
+
+    def compile_source_file_command(self, source_file):
+        """
+        Returns the command to compile a single source file
+        """
+        if source_file.is_vhdl:
+            return self.compile_vhdl_file_command(source_file)
+
+        if source_file.is_any_verilog:
+            return self.compile_verilog_file_command(source_file)
+
+        LOGGER.error("Unknown file type: %s", source_file.file_type)
+        raise CompileError
+
+    def _compile_all_source_files(self, source_files_by_library: Dict[Library, List[SourceFile]], printer):
+        """
+        Compiles all source files and prints status information
+        """
+        try:
```

Comment:
Is the output consistent with the rest of VUnit's reporting?

Reply:
If I had to guess, the answer would be no.